server/plugin/command/create_event.go

```diff
@@ -26,121 +32,100 @@ type userError struct {
-				State:           "Nevada",
-				CountryOrRegion: "US",
-				PostalCode:      "89158",
-			},
-			Coordinates: &remote.Coordinates{
-				Latitude:  47.672,
-				Longitude: -102.103,
-			},
-		},
-		Attendees: attendees,
-		Start: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Format(time.RFC3339),
-		},
-		End: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Add(time.Hour).Format(time.RFC3339),
-		},
-	}
-
-	// parse flags and start overriding Demo Defaults
+	event := &remote.Event{}
+
 	createFlagSet := getCreateEventFlagSet()
 	err := createFlagSet.Parse(args)
 	if err != nil {
 		return event, nil, err
 	}
 
-	//                //
-	// Required Flags //
-	//                //
+	help, err := createFlagSet.GetBool("help")
+	if help == true {
+		return nil, &userError{ErrorMessage: fmt.Sprintf(getCreateEventFlagSet().FlagUsages())}, nil
+	}
+
 	subject, err := createFlagSet.GetString("subject")
 	if err != nil {
 		return event, nil, err
 	}
 	// check that next arg is not a flag "--"
-	if subject == "" || strings.HasPrefix(subject, "--") {
+	if strings.HasPrefix(subject, "--") {
 		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
 	}
 	event.Subject = subject
 
-	//                //
-	// Optional Flags //
-	//                //
+	body, err := createFlagSet.GetString("body")
+	if err != nil {
+		return event, nil, err
+	}
+	// check that next arg is not a flag "--"
+	if strings.HasPrefix(body, "--") {
+		return event, &userError{ErrorMessage: "must specify an event body"}, nil
+	}
+	event.Body = &remote.ItemBody{
+		Content: body,
+	}
+
 	startTime, err := createFlagSet.GetString("starttime")
 	if err != nil {
 		return event, nil, err
 	}
 	if strings.HasPrefix(startTime, "--") {
-		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
+		return event, &userError{ErrorMessage: "must specify an starttime"}, nil
+	}
+	event.Start = &remote.DateTime{
+		DateTime: startTime,
+		TimeZone: "Pacific Standard Time",
 	}
-	event.Start.DateTime = startTime
 
 	endTime, err := createFlagSet.GetString("endtime")
 	if err != nil {
 		return event, nil, err
 	}
 	if strings.HasPrefix(endTime, "--") {
-		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
+		return event, &userError{ErrorMessage: "must specify an endtime"}, nil
+	}
+	event.End = &remote.DateTime{
+		DateTime: endTime,
+		TimeZone: "Pacific Standard Time",
+	}
+
+	allday, err := createFlagSet.GetBool("allday")
+	if err != nil {
+		return event, nil, err
+	}
+	event.IsAllDay = allday
+
+	reminder, err := createFlagSet.GetInt32("reminder")
+	if err != nil {
+		return event, nil, err
+	}
+	if strings.HasPrefix(strconv.Itoa(int(reminder)), "--") {
+		return event, &userError{ErrorMessage: "must specify an reminder"}, nil
+	}
+	event.ReminderMinutesBeforeStart = reminder
+
+	location, err := createFlagSet.GetStringSlice("location")
```

Comment:
Thinking forward here. Gathering 6 parameters (or even trying to pre-fill them using what we know about the user) seems impractical from the UX standpoint. This is begging for an intelligent autocomplete. Is there a Microsoft Graph API we can use for this? Please file a separate Jira ticket on how we should support Location when creating events.

Of particular importance are video calls. There may be some overlap with Location since these are interrelated datum in the UX.

Reply:
added to https://mattermost.atlassian.net/browse/MM-21390